airflow/kubernetes/pod_generator.py

```diff
@@ -301,51 +352,151 @@ def reconcile_pods(base_pod: k8s.V1Pod, client_pod: k8s.V1Pod) -> k8s.V1Pod:
-        merge_objects(base_pod.spec, client_pod_cp.spec)
-        merge_objects(base_pod, client_pod_cp)
+            setattr(client_obj_cp, field_name, appended_fields)
+            return client_obj_cp
+
+        if base_pod.spec and not client_pod.spec:
+            client_pod_cp.spec = base_pod.spec
+        elif client_pod_cp.spec and base_pod.spec:
+            client_container = client_pod_cp.spec.containers[0]
+            base_container = base_pod.spec.containers[0]
+            cc1 = extend_object_field(base_container, client_container, 'volume_mounts')
+            cc2 = extend_object_field(base_container, cc1, 'env')
+            cc3 = extend_object_field(base_container, cc2, 'env_from')
+            cc4 = extend_object_field(base_container, cc3, 'ports')
+            cc5 = extend_object_field(base_container, cc4, 'volume_devices')
+
+            cc6 = merge_objects(base_container, cc5)
+            client_pod_cp.spec.containers[0] = cc6
+            # Just append any additional containers from the base pod
+            client_pod_cp.spec.containers.extend(base_pod.spec.containers[1:])
+            merged_spec = extend_object_field(base_pod.spec, client_pod_cp.spec, 'volumes')
+            client_pod_cp.spec = merge_objects(base_pod.spec, merged_spec)
+
+        client_pod_cp.metadata = merge_objects(base_pod.metadata, client_pod_cp.metadata)
+        client_pod_cp = merge_objects(base_pod, client_pod_cp)
 
         return client_pod_cp
+
+    @staticmethod
+    def construct_pod(
+        dag_id: str,
+        task_id: str,
+        pod_id: str,
+        try_number: int,
+        date: str,
+        command: List[str],
+        kube_executor_config: Optional[k8s.V1Pod],
+        worker_config: k8s.V1Pod,
+        namespace: str,
+        worker_uuid: str
+    ) -> k8s.V1Pod:
+        """
+        Construct a pod by gathering and consolidating the configuration from 3 places:
+            - airflow.cfg
+            - executor_config
+            - dynamic arguments
+        """
+
+        dynamic_pod = PodGenerator(
+            namespace=namespace,
+            labels={
+                'airflow-worker': worker_uuid,
+                'dag_id': dag_id,
+                'task_id': task_id,
+                'execution_date': date,
+                'try_number': str(try_number),
+            },
+            cmds=command,
+            name=pod_id
+        ).gen_pod()
+
+        # Reconcile the pod generated by the Operator and the Pod
+        # generated by the .cfg file
+        pod_with_executor_config = PodGenerator.reconcile_pods(worker_config,
+                                                               kube_executor_config)
+        # Reconcile that pod with the dynamic fields.
+        return PodGenerator.reconcile_pods(pod_with_executor_config, dynamic_pod)
+
+    @staticmethod
+    def deserialize_model_file(api_client: ApiClient, path: str) -> k8s.V1Pod:
```

Comment:
There are two very similar methods:: `deserialize_model_file`, `deserialize_model_string`.  It may be worth joining them and recognizing the names or types of the method's arguments.